seaworthy/logs.py

```diff
@@ -3,6 +3,7 @@
 """
 
 import re
+from abc import ABC, abstractmethod
```

Comment:
I did not know about `abc`

Reply:
`abc` is vastly inferior to `zope.interface`, but I think it's useful here. Also, I've never actually used it before.

Reply:
So why not `zope.interface`?

Reply:
Mostly because the benefits for this don't outweigh the cost of adding another dependency.